main/src/modules/Assembly.scala

```diff
@@ -70,25 +73,33 @@ object Assembly {
     }
   }
 
-  private def classpathIterator(inputPaths: Agg[os.Path]): Agg[(String, InputStream)] =
+  private def classpathIterator(inputPaths: Agg[os.Path], assemblyRules: Seq[Assembly.Rule]): Agg[(String, InputStream)] = {
+    val shadeRules = assemblyRules.collect {
+      case Rule.Relocate(from, to) => ShadePattern.Rename(List(from -> to)).inAll
```

Comment:
I prefer exposing our own API (`Rule.Relocate`) and mapping it to jarjar-abram's `ShadePattern.Rename` in case we are swapping the shading lib.